File: 2120-execution-of-all-suffix-instructions-staying-in-a-grid/2120-execution-of-all-suffix-instructions-staying-in-a-grid.cpp
class Solution {
public:
    vector<int> executeInstructions(int n, vector<int>& startPos, string s) {
        vector<int> ans;
        int count = 0;
        int ini = startPos[0];
        int inj = startPos[1];
        for(int  i = 0; i < s.length(); i++){
            for(int  j = i; j < s.length(); j++){
                if(s[j] == 'R') startPos[1]++;
                else if(s[j] == 'L') startPos[1]--;
                else if(s[j] == 'U') startPos[0]--;
                else if(s[j] == 'D') startPos[0]++;
                
                if(startPos[1] > -1 && startPos[1] < n && startPos[0] > -1 && startPos[0] < n) count++;
                else j = INT_MAX-1;
            }
            ans.push_back(count);
            count = 0;
            startPos = {ini, inj};
        }
        return ans;
    }
};

File: 2120-execution-of-all-suffix-instructions-staying-in-a-grid/2120-execution-of-all-suffix-instructions-staying-in-a-grid.py
class Solution:
    def executeInstructions(self, n: int, startPos: List[int], s: str) -> List[int]:
        li = []
        count = 0;
        ini = startPos[0]
        inj = startPos[1]
        for i in range(len(s)):
            for j in range(i, len(s)):
                if s[j] == 'R' : startPos[1] += 1
                elif s[j] == 'L' : startPos[1] -= 1
                elif s[j] == 'U' : startPos[0] -= 1
                elif s[j] == 'D' : startPos[0] += 1
                
                if (startPos[1] > -1 and startPos[1] < n and startPos[0] > -1 and startPos[0] < n) : count += 1
                else : startPos = [-1,-1] 
            startPos = [ini, inj]
            li.append(count)
            count = 0
        return li